lib/common/mobile/android/logcat-helper.ts

```diff
@@ -108,39 +147,92 @@ export class LogcatHelper implements Mobile.ILogcatHelper {
 	}
 
 	private forceStop(deviceIdentifier: string): void {
-		this.mapDevicesLoggingData[
-			deviceIdentifier
-		].loggingProcess.removeAllListeners();
-		this.mapDevicesLoggingData[deviceIdentifier].loggingProcess.kill("SIGINT");
-		this.mapDevicesLoggingData[
-			deviceIdentifier
-		].lineStream.removeAllListeners();
+		const loggingData = this.mapDevicesLoggingData[deviceIdentifier];
+		loggingData.loggingProcess?.removeAllListeners();
+		loggingData.loggingProcess?.kill("SIGINT");
+		loggingData.lineStream?.removeAllListeners();
+
+		loggingData.appStartTrackingProcess?.kill("SIGINT");
+		loggingData.lineStream?.removeAllListeners();
+
 		delete this.mapDevicesLoggingData[deviceIdentifier];
 	}
 
-	private async getLogcatStream(deviceIdentifier: string, pid?: string) {
+	/**
+	 * @deprecated - we likely don't need this anymore, and can simplify the code...
+	 */
+	private async isLogcatPidSupported(deviceIdentifier: string) {
 		const device = await this.$devicesService.getDevice(deviceIdentifier);
 		const minAndroidWithLogcatPidSupport = "7.0.0";
-		const isLogcatPidSupported =
+		return (
 			!!device.deviceInfo.version &&
 			semver.gte(
 				semver.coerce(device.deviceInfo.version),
 				minAndroidWithLogcatPidSupport
-			);
+			)
+		);
+	}
+
+	private async getLogcatStream(deviceIdentifier: string, pid?: string) {
+		const isLogcatPidSupported = await this.isLogcatPidSupported(
+			deviceIdentifier
+		);
 		const adb: Mobile.IDeviceAndroidDebugBridge = this.$injector.resolve(
 			DeviceAndroidDebugBridge,
 			{ identifier: deviceIdentifier }
 		);
-		const logcatCommand = ["logcat"];
+
+		// -T 1 - shows only new logs after starting adb logcat
+		const logcatCommand = ["logcat", "-T", "1"];
+
+		const acceptedTags = [
+			"chromium",
+			'"Web Console"',
+			"JS",
+			"System.err",
+			"TNS.Native",
+			"TNS.Java",
+		];
 
 		if (pid && isLogcatPidSupported) {
 			logcatCommand.push(`--pid=${pid}`);
+
+			acceptedTags.forEach((tag) => {
+				// -s <tag> - shows only logs with the specified tag
+				logcatCommand.push("-s", tag);
+			});
 		}
+
 		const logcatStream = await adb.executeCommand(logcatCommand, {
 			returnChildProcess: true,
 		});
+
 		return logcatStream;
 	}
+
+	private async getAppStartTrackingLogcatStream(
+		deviceIdentifier: string,
+		appId?: string
+	) {
+		const adb: Mobile.IDeviceAndroidDebugBridge = this.$injector.resolve(
+			DeviceAndroidDebugBridge,
+			{ identifier: deviceIdentifier }
+		);
+
+		// -b system  - shows the system buffer/logs only
+		// -T 1 			- shows only new logs after starting adb logcat
+		const logcatCommand = [`logcat`, `-b`, `system`, `-T`, `1`];
```

Comment:
Note: Added the `-b system` flag. Probably not a huge difference, but only showing the `system` buffer should theoretically be faster than showing the default (system, crash, main).